src/infer_stats.cc

```diff
@@ -210,6 +210,132 @@ InferenceStatsAggregator::UpdateSuccessCacheMiss(
 #endif  // TRITON_ENABLE_METRICS
 }
 
+Status
+InferenceStatsAggregator::UpdateResponseSuccess(
+    const std::string& key, const uint64_t response_start_ns,
+    const uint64_t compute_output_start_ns, const uint64_t response_end_ns)
+{
+  if (response_start_ns > compute_output_start_ns) {
+    return Status(
+        Status::Code::INVALID_ARG,
+        "Response start cannot happen after compute output start");
```

Comment:
Should we make it more descriptive? Something like:
"Unexpected behavior: response start timestamp was recorded after compute output start"

Reply:
I don't know if this is cleaner, because the user could have recorded the `response start` first and then followed by the `compute output start`, but reported the `compute output start` as the `response start` and `response start` as the `compute output start`. Then, the message could send the user looking at the wrong place, which is why I want to avoid speculating on what the user did, but simply say what is passed into the API.

Maybe we can write
```
Response start cannot happen after compute output start, check response statistics timestamps.
```
?

Reply:
We ca keep it as is then=)